### PR TITLE
[o-mr0] changes for FBE (part 1/2)

### PR DESCRIPTION
**Pull request by jamuir**
changes to support verity and FBE.

### CODE REVIEW

**Comment by jamuir** [comment]
there are also changes needed in device-sony-common:

https://github.com/sonyxperiadev/device-sony-common/pull/391

File: platform.mk [hunk 1]
```diff
@@ -150,3 +150,11 @@ PRODUCT_PROPERTY_OVERRIDES += \
 PRODUCT_PROPERTY_OVERRIDES += \
     sys.usb.controller=a800000.dwc3 \
     sys.usb.rndis.func.name=gsi
+
+# verity
+PRODUCT_COPY_FILES += \
+    frameworks/native/data/etc/android.software.verified_boot.xml:system/etc/permissions/android.software.verified_boot.xml
```
Comment (bartcubbins) [inline platform.mk]: I think it must be moved to vendor

Comment (jamuir) [reply platform.mk]: why?

putting the xml file under `system/etc/permissions` is how it is done on bullhead and other devices:

```bash
android-8/device$ mgrep verified_boot

./google/dragon/device.mk:102:    frameworks/native/data/etc/android.software.verified_boot.xml:system/etc/permissions/android.software.verified_boot.xml \
./huawei/angler/device.mk:152:    frameworks/native/data/etc/android.software.verified_boot.xml:system/etc/permissions/android.software.verified_boot.xml \
./lge/bullhead/device.mk:120:    frameworks/native/data/etc/android.software.verified_boot.xml:system/etc/permissions/android.software.verified_boot.xml \
```

**Comment by jerpelea** [comment]
in android 8 er are migrating all configuration files to vendor

**Comment by jamuir** [comment]
Are you doing that to support treble?

On Dec 20, 2017 6:09 PM, "Alin Jerpelea" <notifications@github.com> wrote:

in android 8 er are migrating all configuration files to vendor

—
You are receiving this because you authored the thread.

Reply to this email directly, view it on GitHub
<https://github.com/sonyxperiadev/device-sony-yoshino/pull/70#issuecomment-353209664>,
or mute the thread

**Comment by cryptomilk** [comment]
Note that if you turn on FBE, then the sdcard is not seen by apps anymore. I'm currently using the following property as a workaround:

```
# Make sdcards visibile in StorageManager                                                                                                                              
persist.sys.adoptable_fbe=true
```

**Comment by jamuir** [comment]
Thanks for pointing that out.

We have FBE enabled in our in-house builds, but I don't know if we ever tested sdcard functionality with it (however, that was with Nougat).

btw, is your sdcard formatted with a "fat" file-system?

There is a warning at https://source.android.com/security/encryption/file-based stating that FBE is not compatible with adoptable storage.  Perhaps there is some internal check for this and the logic isn't quite right.

**Comment by cryptomilk** [comment]
Yes, with FBE support enabled the sdcard is not adoptable! Normally you're asked it you want to use it as portable sdcard or adopt it.

However, if it isn't adoptable it isn't marked as visible either. If it isn't visible you can't use it as portable storage so that apps and the mediaextractor can see it. So without forcing the sdcard to be adoptable it is pretty much useless.

**Comment by cryptomilk** [comment]
Currently I have my sdcard exfat formatted.

**Comment by jerpelea** [comment]
please rebase and push to 8.1

**Pull request closed without merging**